docs/wmemcheck.md

```diff
@@ -0,0 +1,8 @@
+
+Wmemcheck provides debug output for invalid mallocs, reads, and writes.
+
+How to use:
+1. When building Wasmtime, add the CLI flag "--features wmemcheck" to compile with wmemcheck configured.
+    > cargo build --features wmemcheck
+2. When running your wasm module, add the CLI flag "--wmemcheck".
+    > wasmtime run --wmemcheck test.wasm
```

Comment:
Let's add a brief note that wmemcheck requires a non-stripped wasm (with a name section) so that it can recognize the malloc and free functions, and doesn't work with custom allocators yet.

Also, can we add a reference to Valgrind? Something like "wmemcheck is inspired by, and replicates the basic functionality of, Valgrind's memcheck tool."